### ISSUE TITLE
Conflict handler not running on conflict

### ISSUE THREAD

**Issue by pokey** [aws-amplify/amplify-js#5945]
**Describe the bug**
When the server reports a ConflictUnhandled error in response to a mutation from DataStore, it provides a `null` field in `error.data`.  Here is the `error` object as captured from a breakpoint on https://github.com/aws-amplify/amplify-js/blob/67fac50cd734338ac76797d06111fc5ca911bd48/packages/datastore/src/sync/processors/mutation.ts#L248:

```json
{
  "path": [
    "updateFlashcard"
  ],
  "data": null,
  "errorType": "ConflictUnhandled",
  "errorInfo": null,
  "locations": [
    {
      "line": 2,
      "column": 3,
      "sourceName": null
    }
  ],
  "message": "Conflict resolver rejects mutation."
}
```

Thus when `error.data` is passed to `modelInstanceCreator`, there is an error trying to add `null` to a `WeakSet`, so the following error is reported:

```
backend.js:32 [WARN] 23:19.287 DataStore - conflict trycatch TypeError: Invalid value used in weak set
    at WeakSet.add (<anonymous>)
    at MutationProcessor.modelInstanceCreator (datastore.js:204)
    at MutationProcessor.<anonymous> (mutation.js:237)
    at step (tslib.js:139)
    at Object.throw (tslib.js:120)
    at rejected (tslib.js:111)
    at tryCallOne (core.js:37)
    at core.js:123
    at JSTimers.js:277
    at _callTimer (JSTimers.js:135)
```

So then my custom local conflict handler is never run.  But weak set issue aside, looks like if `error.data` is `null`, there's no way for you to construct the `remoteModel`, so my conflict handler won't be particularly useful.

**To Reproduce**

1. Create a data model using optimistic concurrency
1. Create a conflict locally, eg by writing to the same object quickly twice in a row

**Expected behavior**
My conflict handler should run

**Code Snippet**

**Screenshots**

**What is Configured?**

<details>
  <summary><strong>Environment</strong></summary>

```
npx envinfo --system --binaries --browsers --npmPackages --npmGlobalPackages

  System:
    OS: macOS Mojave 10.14.5
    CPU: (8) x64 Intel(R) Core(TM) i7-6920HQ CPU @ 2.90GHz
    Memory: 870.25 MB / 16.00 GB
    Shell: 5.7.1 - /usr/local/bin/zsh
  Binaries:
    Node: 12.16.1 - ~/n/bin/node
    Yarn: 1.22.0 - /usr/local/bin/yarn
    npm: 6.13.4 - ~/n/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  npmPackages:
    @aws-amplify/core: ^3.2.1 => 3.2.4 
    @aws-amplify/datastore: ^2.0.5 => 2.0.8 
    @babel/core: ^7.9.0 => 7.9.0 
    @babel/runtime: ^7.9.2 => 7.9.2 
    @react-native-community/eslint-config: ^1.0.0 => 1.1.0 
    @react-native-community/netinfo: ^5.7.0 => 5.7.1 
    amazon-cognito-identity-js: ^4.2.1 => 4.2.1 
    aws-amplify: ^3.0.8 => 3.0.8 
    aws-amplify-react-native: ^4.0.2 => 4.0.4 
    babel-jest: ^25.2.6 => 25.3.0 
    eslint: ^6.8.0 => 6.8.0 
    immutability-helper: ^3.0.2 => 3.0.2 
    ini: ^1.3.5 => 1.3.5 
    inquirer: ^6.5.1 => 6.5.2 
    jest: ^25.2.7 => 25.3.0 
    jstat: ^1.9.2 => 1.9.2 
    lodash: ^4.17.15 => 4.17.15 
    metro-react-native-babel-preset: ^0.59.0 => 0.59.0 
    react: 16.11.0 => 16.11.0 
    react-native: 0.62.1 => 0.62.1 
    react-native-deck-swiper: ^1.6.7 => 1.6.7 
    react-native-get-random-values: ^1.3.1 => 1.3.1 
    react-native-uuid: ^1.4.9 => 1.4.9 
    react-native-vector-icons: ^6.6.0 => 6.6.0 
    react-native-view-overflow: ^0.0.5 => 0.0.5 
    react-test-renderer: 16.11.0 => 16.11.0 
    react-use: ^14.1.1 => 14.1.1 
    use-deep-compare-effect: ^1.3.1 => 1.3.1 
  npmGlobalPackages:
    @aws-amplify/cli: 4.17.2
    npm: 6.13.4
    react-devtools: 4.6.0
    react-native-cli: 2.0.1
```

</details>

**Smartphone (please complete the following information):**

**Additional context**

**_You can turn on the debug mode to provide more info for us by setting window.LOG_LEVEL = 'DEBUG'; in your app._**

**Comment by diamanthaxhimusa** [comment]
The same thing is happening to me!

**Comment by iartemiev** [comment]
@pokey thank you for identifying the issue and providing reproduction steps. We've opened a PR to resolve this. I will update this ticket once it's merged.

**Comment by iartemiev** [comment]
The fix has been released as part of **@aws-amplify/datastore@2.2.5**
